misc/registrar_validation/test.py

```diff
@@ -0,0 +1,364 @@
+# pylint: skip-file
+# because this is an old partial file
+
+from collections.abc import Callable
+import copy
+import datetime
+import glob
+import json
+import json
+import os
+from pathlib import Path
+import random
+import re
+import shutil
+import shutil
+import signal
+import ssl
+import subprocess
+import sys
+import time
+import time
+from typing import Any
+from typing import Iterator
+import jwt
+import paho.mqtt.client as mqtt
+import pytest
+
+KEYS_TO_REDACT = ["timestamp", "version", "operation"]
+REDACTED_VALUE = "redacted"
+
+SITE_PATH = os.getenv("SITE_PATH") if os.getenv("SITE_PATH") else "."
+devices_list = [
+    x.parent.stem
+    for x in Path(SITE_PATH).glob(
+        os.path.join("udmi/devices/*/rsa_private.pem")
+    )
+]
+
+
+class MqttClient:
+  algorithm = "RS256"
+  mqtt_bridge_port = 443
+  mqtt_bridge_hostname = "mqtt.bos.goog"
+  jwt_exp_mins = 20
+  puback_regex = r"^Received PUBACK \(Mid: (\d+)\)"
+  topic_regex = r"^/devices/(.*)/config"
+
+  def __init__(self, site_path, project_id, device_id):
+    self.pubacks = []
+    self.device_id = device_id
+    self.project_id = project_id
+    self.private_key_file = os.path.join(
+        site_path, "devices", device_id, "rsa_private.pem"
+    )
+
+    with open(
+        os.path.join(site_path, "cloud_iot_config.json"), encoding="utf-8"
+    ) as f:
+      cloud_iot_config = json.load(f)
+
+    self.registry_id = cloud_iot_config.get("registry_id")
+    self.cloud_region = cloud_iot_config.get("cloud_region")
+
+    self.configs = {}
+
+    self.client = self.start_client()
+
+  def subscribe_proxy_config(self, proxy_id):
+    self.client.subscribe(f"/devices/{proxy_id}/config", qos=1)
+
+  def attach_device(self, device_id) -> int:
+    """Returns mid"""
+    attach_topic = f"/devices/{device_id}/attach"
```

Comment:
We should talk about your general approach here, because although it's "quick and dirty" to crank out tools like this, it doesn't contribute as much in the long run to the overall UDMI ecosystem.  In general with software development our goal is to amortize costs in order to solve problems _efficiently_ in the long run, not just crank out "solutions"!